contracts/proxies/SafeProxy.sol

```diff
@@ -31,21 +31,25 @@ contract SafeProxy {
 
     /// @dev Fallback function forwards all transactions and returns all received return data.
     fallback() external payable {
-        // solhint-disable-next-line no-inline-assembly
+        /* solhint-disable no-inline-assembly */
+        /// @solidity memory-safe-assembly
         assembly {
             let _singleton := and(sload(0), 0xffffffffffffffffffffffffffffffffffffffff)
             // 0xa619486e == keccak("masterCopy()"). The value is right padded to 32-bytes with 0s
             if eq(calldataload(0), 0xa619486e00000000000000000000000000000000000000000000000000000000) {
                 mstore(0, _singleton)
                 return(0, 0x20)
             }
-            calldatacopy(0, 0, calldatasize())
-            let success := delegatecall(gas(), _singleton, 0, calldatasize(), 0, 0)
-            returndatacopy(0, 0, returndatasize())
+            let ptr := mload(0x40)
```

Comment:
Does it have any gas implications?

Reply:
Probably will have a slight gas implication of reading from the memory, haven't checked explicitly though.

Reply:
We have benchmarks in the repo that are rather easy to run

Reply:
At max, around 30 gas higher, on average I think around 18 gas higher (based on a manual look on the results of `benchmark`)

Reply:
That sounds decent. Thanks